shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1924,6 +1930,36 @@ - (void)pressesCancelled:(NSSet<UIPress*>*)presses
   [_textInputPlugin.get().viewController pressesCancelled:presses withEvent:event];
 }
 
+#pragma mark - Undo/Redo support
+
+- (void)ensureUndoEnabled API_AVAILABLE(ios(9.0)) {
+  if (![self.undoManager canUndo]) {
+    self.undoManager.groupsByEvent = NO;
+    [self.undoManager beginUndoGrouping];
+    [self.undoManager registerUndoWithTarget:self
+                                     handler:^(id target){
+
+                                     }];
+    [self.undoManager endUndoGrouping];
+    self.undoManager.groupsByEvent = YES;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.01 * (double)NSEC_PER_SEC),
+                   dispatch_get_main_queue(), ^{
+                     [self.undoManager removeAllActionsWithTarget:self];
+                   });
+  }
+}
+
+- (void)setUndoManager:(NSUndoManager*)undoManager {
+  self.undoManagerOverride = undoManager;
```

Comment:
why caches this?

Reply:
So that I can set a mock object for tests – maybe there's a better way if I could mock the `FlutterTextInput` itself instead?

Reply:
Maybe just stub the method? https://ocmock.org/reference/#stubing-methods

Reply:
Anyhow you don't need to expose this as a public method. A private category in the test should do it.